### ISSUE TITLE
Crash, when source fragment in error message contains non-ascii characters

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#767]
When sphinx-build is invoked with -W, then following mistakes in source cause sphinx to crash instead of displaying nice error message:

Missing empty line:

```
.. sourcecode:: python
    for i in range(a,b):
        # käsud 
```

Too short section underline:

```
Lisanäited
------ 
```

Stacktrace:

```
Warning, treated as error:
Traceback (most recent call last):
  File "c:\python27\scripts\sphinx-build-script.py", line 8, in
<module>
    load_entry_point('Sphinx==1.0.7', 'console_scripts', 'sphinx-
build')()
  File "c:\python27\lib\site-packages\sphinx-1.0.7-py2.7.egg\sphinx
\__init__.py"
, line 67, in main
    return cmdline.main(argv)
  File "c:\python27\lib\site-packages\sphinx-1.0.7-py2.7.egg\sphinx
\cmdline.py",
 line 212, in main
    print >>error, err
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe4' in
position 202
: ordinal not in range(128) 
```

Workaround: change line 212 in cmdline.py from

```
print >>error, err
```

to 

```
print >>error, err.message.encode("ascii", "backslashreplace") 
```

If sphinx-build is invoked without -W, then error message is displayed fine. 

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/767
- Originally reported by: [aivarannamaa](https://bitbucket.org/aivarannamaa)
- Originally created at: 2011-09-19T17:16:41.953

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-21 08:45:19+00:00_

Fix #767: safely encode SphinxErrors when printing to sys.stderr.

→ <<cset c05063d0a5a0e873e03561bbe78b78dd73df80a1>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-21 08:47:52+00:00_

Fix #767: safely encode SphinxErrors when printing to sys.stderr.

→ <<cset 19775589e9de7855fe48ed94b0c580537db62820>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:32:19+00:00_

Removing version: 1.0.7 (automated comment)
